src/commands/roleCommands/changecolor.ts

```diff
@@ -283,25 +284,22 @@ export class ColorRoles {
           position: rolePosition,
           mentionable: false,
         }))
+      await member.roles.add(colorRole).catch(console.error)
+    }
 
-      // Remove and delete existing role if exists
-      const existingRole = member.roles.cache.find((role) => ColorRoles.hexExp.test(role.name))
-      return member.roles.add(colorRole).then(async (guildMember) => {
-        if (existingRole) {
-          member.roles.remove(existingRole).catch(console.error)
-          const roleToDelete = guild.roles.cache.find((role) => role.id === existingRole.id)
-          if (
-            roleToDelete &&
-            (roleToDelete.members.size === 0 ||
-              (roleToDelete.members.size === 1 && roleToDelete.members.some((_, id) => id === member.id)))
-          ) {
-            guild.roles.delete(existingRole.id).catch(console.error)
-          }
-        }
-
-        return guildMember
-      })
+    if (existingRole) {
```

Comment:
We should check if the old color role has the same name as the new color otherwise doing the same command back to back or losing twice in a row will remove the color.

Reply:
Good call, added a name check